qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,807 @@
+    txids = [(run, z_mergetoaddress(results, run[0], zcash, run[1], run[2])) for run in runs]
+    # Then wait for balance updates caused by successful transactions
+    return [
+        wait_and_check_balance(results, run[0], zcash, run[2], run[3]) if txid is not None else 0
+        for (run, txid) in txids]
+
+
+#
+# Test runners
+#
+
+def simple_commands(zcash):
+    results = {}
+    run_cmd(results, '2a', zcash, 'getinfo'),
+    run_cmd(results, '2b', zcash, 'help'),
+    run_cmd(results, '5o', zcash, 'getwalletinfo'),
+    run_cmd(results, '6a', zcash, 'getpeerinfo'),
+    run_cmd(results, '6b', zcash, 'getnetworkinfo'),
+    run_cmd(results, '6c', zcash, 'getdeprecationinfo'),
+    run_cmd(results, '6d', zcash, 'getconnectioncount'),
+    run_cmd(results, '6e', zcash, 'getaddednodeinfo', [False]),
+    run_cmd(results, '7a', zcash, 'getblocksubsidy'),
+    run_cmd(results, '7c', zcash, 'getmininginfo'),
+    run_cmd(results, '7d', zcash, 'getnetworkhashps'),
+    run_cmd(results, '7e', zcash, 'getnetworksolps'),
+    return results
+
+def transaction_chain(zcash):
+    results = {}
+
+    # Generate the various addresses we will use
+    sprout_zaddr_1 = run_cmd(results, '3a', zcash, 'z_getnewaddress', ['sprout'])
+    sprout_zaddr_2 = run_cmd(results, '3b', zcash, 'z_getnewaddress', ['sprout'])
+    sprout_zaddr_3 = run_cmd(results, '3b', zcash, 'z_getnewaddress', ['sprout'])
+    taddr_1 = run_cmd(results, '3c', zcash, 'getnewaddress')
+    taddr_2 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    taddr_3 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    taddr_4 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    taddr_5 = run_cmd(results, '3d', zcash, 'getnewaddress')
+    sapling_zaddr_1 = run_cmd(results, '3e', zcash, 'z_getnewaddress', ['sapling'])
+    sapling_zaddr_2 = run_cmd(results, '3f', zcash, 'z_getnewaddress', ['sapling'])
+    sapling_zaddr_3 = run_cmd(results, '3f', zcash, 'z_getnewaddress', ['sapling'])
+
+    # Check that the zaddrs are all listed
+    zaddrs = run_cmd(results, '5a', zcash, 'z_listaddresses')
+    if (sprout_zaddr_1 not in zaddrs or
+            sprout_zaddr_2 not in zaddrs or
+            sapling_zaddr_1 not in zaddrs or
+            sapling_zaddr_2 not in zaddrs):
+        results['5a'] = False
+
+    # Validate the addresses
+    ret = run_cmd(results, '5b', zcash, 'z_validateaddress', [sprout_zaddr_1])
+    if not ret['isvalid'] or ret['type'] != 'sprout':
+        results['5b'] = False
+
+    ret = run_cmd(results, '5c', zcash, 'z_validateaddress', [sapling_zaddr_1])
+    if not ret['isvalid'] or ret['type'] != 'sapling':
+        results['5c'] = False
+
+    # Set up beginning and end of the chain
+    print('#')
+    print('# Initialising transaction chain')
+    print('#')
+    print()
+    chain_end = input('Type or paste transparent address where leftover funds should be sent: ')
+    if not zcash.validateaddress(chain_end)['isvalid']:
```

Comment:
I'm running into issues getting past this point. When I give it a valid transparent address, it says it's invalid, which I think is happening because I'm testing with a very old `.zcash` directory and it needs to rescan/redownload blocks, so the RPC is failing (this was wrong, see below). It would be nice if it made the real error visible.

Update: actually what was happening was I was trying to give it a mainnet address without passing the `--mainnet` flag. If it said "Type or paste a (mainnet|testnet) transparent address where leftover...", conditioned on the `--mainnet` flag, I might have realized my error sooner.

I can get past this point now but my `zcashd` is redownloading the entire chain so presumably I have to wait for that to finish before I can continue testing.